src/org/usfirst/frc/team930/robot/controller/BBSController.java

```diff
@@ -0,0 +1,123 @@
+package org.usfirst.frc.team930.robot.controller;
+
+import edu.wpi.first.wpilibj.SpeedController;
+
+/**
+ * Concrete class for the implementation of the Bang Bang Shooter Controller,
+ * which will set the motor speed to max speed if the motor runs below the
+ * target RPM and will let it run freely otherwise (ie, speed of zero).
+ * <p>
+ * For best results, tune your acceleration values of the speed controller so
+ * that it doesn't jump around the set point - note, it will still control to
+ * your set points, but it will consistently overaccelerate in the calculation
+ * timeframe. As a reference, accel = 1 will produce ~3800 RPM
+ * <p>
+ * Based on the implementation on <a
+ * href="http://www.chiefdelphi.com/media/papers/2663">Chief Delphi</a>.
+ * 
+ * @see Controller.java
+ * @author Nick Janovetz
+ * @version 2016.02.03_3 Added finality to m_sc
+ */
+public class BBSController extends Controller {
+
+	private static int count = 0;
+
+	private ControllerSource m_source;
+	private final SpeedController m_sc;
+	private double m_targetRPM;
+	private double m_accel;
+	private int m_count;
+
+	/**
+	 * Constructs a Controller given an RPM input, target, speedcontroller, and
+	 * acceleration
+	 * <p>
+	 * The speedcontroller must be in coast mode - check your NI Dashboard
+	 * 
+	 * @param source
+	 *            The PID input for the motor that returns RPM
+	 * @param sc
+	 *            The SpeedController that will be controlled
+	 * @param targetRPM
+	 *            The target RPM for the SpeedController to spin at
+	 * @param accel
+	 *            The amount the speedcontroller impulses to when below target,
+	 *            0 <= accel <= 1
+	 */
+	public BBSController(ControllerSource source, SpeedController sc,
+			double targetRPM, double accel) {
+		m_source = source;
+		m_sc = sc;
+		m_targetRPM = targetRPM;
+		m_accel = accel < 0f ? 0f : (accel > 1f ? 1 : accel);
```

Comment:
Try to avoid nested ternary statements. Difficult to read. 1 line if-statements are typically easier to read.

`if(accel < 0) m_accel = 0;`